bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/ProjectPerTargetProvisioningStrategy.java

```diff
@@ -78,6 +81,27 @@ public Map<BazelProject, Collection<ClasspathEntry>> computeClasspaths(Collectio
 
             var workspaceRoot = workspace.getLocation().toPath();
 
+            var importDepth = workspace.getBazelProjectView().importDepth();
+            Set<BazelLabel> availableDependencies = Set.of();
+            if (importDepth > 0) {
+                var targetLabels = targetsToBuild.stream().map(BazelLabel::toString).collect(joining(" + "));
+                availableDependencies = workspace.getCommandExecutor()
+                        .runQueryWithoutLock(
+                            new BazelQueryForLabelsCommand(
+                                    workspace.getLocation().toPath(),
+                                    format(
+                                        "kind(java_library, deps(%s, %d)) + kind(java_import, deps(%s, %d))",
+                                        targetLabels,
+                                        importDepth,
+                                        targetLabels,
+                                        importDepth),
+                                    true,
+                                    format("Querying for depdendencies for projects: %s", targetLabels)))
+                        .stream()
+                        .map(BazelLabel::new)
+                        .collect(toSet());
+            }
```

Comment:
Can this be moved into the parent base class? It seems to be shared between both strategies.